host: Add realm icon in attached cards

backspace opened the pull request:
<img width="386" alt="Boxel 2024-03-11 14-40-07" src="https://github.com/cardstack/boxel/assets/43280/dc1e47fe-f156-42a1-ae5e-ed025c060df4">

This necessitated adding the ability to not render card chrome so I could insert the icon within the border. A box component now accepts `@displayContainer`, it defaults to `true` but we use `false` on the attached cards in atom format to include the icon.

backspace committed:
Remove card boundaries

This will probably break things visually.

backspace committed:
Add assertion for card realm icon

backspace committed:
Add ability to turn off card container

backspace committed:
Add missing property to signature

backspace committed:
Merge branch 'main' into host/ai-message-attached-card-realm-icon-cs-6551

backspace committed:
Remove redundant property

backspace committed:
Add import error expectation

backspace committed:
Merge branch 'main' into host/ai-message-attached-card-realm-icon-cs-6551

backspace committed:
Add selector to fix specificity

We want the extra padding for the delete icon.

burieberry commented on packages/host/app/components/ai-assistant/message/index.gts:
```diff
@@ -240,3 +244,55 @@ export class AiAssistantConversation extends Component<AiAssistantConversationSi
+  };
+}
+
+export class CardPill extends Component<CardPillSignature> {
```
There's actually another `CardPill` component in `components/ai-assistant/card-picker/index`. That is the one we're using when cards are attached in the chat input area of the ai-assistant panel. It would be good if we could consolidate them. The difference would be to not to display the remove button. The styles should be the same otherwise.

backspace replied:
👍🏻 thanks, I’ve merged them now successfully 🎉

<img width="387" alt="Boxel 2024-03-14 11-06-46" src="https://github.com/cardstack/boxel/assets/43280/cc33f7b2-3421-4213-9ddc-fd0c3e358680">

backspace committed:
Remove unused @index property

I couldn’t find anything referencing this in tests… it
could be attached with splattributes anyway?

backspace committed:
Extract shared CardPill component

I’m having weird type failures locally…!

backspace committed:
Fix selectors for previously-attached messages

backspace committed:
Merge branch 'main' into host/ai-message-attached-card-realm-icon-cs-6551

backspace committed:
Merge branch 'main' into host/ai-message-attached-card-realm-icon-cs-6551

backspace committed:
Fix message test selector

backspace committed:
Update selector name for clarity

backspace committed:
Remove miscommitted only

backspace commented:
@cardstack/boxel-developers Luke suggested that more people check this out as the `@displayContainer` change is far-reaching 🤓

habdelra commented on packages/base/field-component.gts:
```diff
@@ -89,13 +90,19 @@ export function getBoxComponent(
+    Args: { format?: Format; displayContainer?: boolean };
+  }> = <template>
+    {{#let (lookupFormat @format) as |f|}}
+      {{#let
```
You can yield 2 (or more) vars in a single {{#let}} instead of nesting them.  there are a couple examples in the code base of this if you search around.

backspace replied:
Fixed now, thanks.

habdelra reviewed (approved):
Looks good, just minor note about unnecessary {{let}} nesting

backspace committed:
Remove nested let
